### PR TITLE
saved queries support tags

### LINKED ISSUE #6711: [Core] Saved queries support `tags`
At a minimum, update body to include a link to the page on docs.getdbt.com requiring updates and what part(s) of the page you would like to see updated.
Originating from this issue: https://github.com/dbt-labs/dbt-core/issues/11155
Originating from this PR: https://github.com/dbt-labs/dbt-core/pull/10987

### PR DESCRIPTION
**Pull request by mirnawong1**
this pr adds a new feature where saved queries supports `tags`.

currently waiting for eng update on this as still working through

[notion docs project](https://www.notion.so/dbtlabs/Semantic-Layer-MF-meta-and-tag-updates-12ebb38ebda780fdb97ec44de2df036a?pvs=4)

Resolves #6711

---
🚀 Deployment available! Here are the direct links to the updated files:

- https://docs-getdbt-com-git-add-tags-dbt-labs.vercel.app/docs/build/saved-queries
- https://docs-getdbt-com-git-add-tags-dbt-labs.vercel.app/docs/dbt-versions/release-notes
- https://docs-getdbt-com-git-add-tags-dbt-labs.vercel.app/docs/use-dbt-semantic-layer/exports
- https://docs-getdbt-com-git-add-tags-dbt-labs.vercel.app/reference/resource-configs/tags

### CODE REVIEW

File: website/docs/docs/build/saved-queries.md [hunk 1]
```diff
@@ -17,8 +17,32 @@ To create a saved query, refer to the following table parameters.
 :::tip
 Note that we use the double colon (::) to indicate whether a parameter is nested within another parameter. So for example, `query_params::metrics` means the `metrics` parameter is nested under `query_params`.
 :::
+
+<!-- For versions 1.9 and higher -->
+<VersionBlock firstVersion="1.9">
```
Comment (theyostalservice) [inline website/docs/docs/build/saved-queries.md:22]: I didn't review this as I'm not sure what has changed and it doesn't seem related to tags

File: website/docs/reference/resource-configs/tags.md [hunk 2]
```diff
@@ -16,59 +16,93 @@ datatype: string | [string]
...
 
-```yml
-version: 2
+The following examples show how to add tags to dbt resources in YAML files. Replace `resource_type` with `models`, `snapshots`, `seeds`, or `saved_queries` as appropriate.
+</VersionBlock>
 
-models:
-  - name: model_name
-    config:
-      tags: <string> | [<string>]
```
Comment (theyostalservice) [inline website/docs/reference/resource-configs/tags.md:48]: this change is wrong, at least for saved queries.  They support both string and [string] types for tags.

File: website/docs/reference/resource-configs/tags.md [hunk 3]
```diff
@@ -164,10 +198,51 @@ seeds:
...
+
+```yaml
+saved_queries:
+  - name: test_saved_query
+    description: "{{ doc('saved_query_description') }}"
+    label: Test saved query
+    config:
+      tags: 
+        - order_metrics
+        - hourly
```
Comment (theyostalservice) [inline website/docs/reference/resource-configs/tags.md:248]: Putting the tags inside the config is not how I built this to work or expected it to work, but it DOES work, so that's pretty cool, and I think it's more in line with how other things work, so... thanks for making this better I guess! lol

File: website/docs/reference/resource-configs/tags.md [hunk 4]
```diff
@@ -16,59 +16,93 @@ datatype: string | [string]
...
     columns:
       - name: column_name
-        tags: [<string>]
+        tags: string
         tests:
-          <test-name>:
+          test-name:
             config:
-              tags: <string> | [<string>]
+              tags: string
```
Comment (theyostalservice) [inline website/docs/reference/resource-configs/tags.md]: I don't know about tags for sub-properties like columns, but since the term is plural, I'd be a little surprised if this change was correct. :/

File: website/docs/reference/resource-configs/tags.md [hunk 5]
```diff
@@ -16,59 +16,93 @@ datatype: string | [string]
...
+The following examples show how to add tags to dbt resources in YAML files. Replace `resource_type` with `models`, `snapshots`, or `seeds` as appropriate.
+</VersionBlock>
+
+<File name='resource_type/properties.yml'>
+
+```yaml
+resource_type:
+  - name: resource_name
+    config:
+      tags: string
```
Comment (theyostalservice) [inline website/docs/reference/resource-configs/tags.md]: (see comment on the red side)

**Review by theyostalservice — COMMENTED** [review]
Thanks for all your work on this, Mirna!

File: website/docs/docs/build/saved-queries.md (hunk 1 above)
Comment (mirnawong1) [reply website/docs/docs/build/saved-queries.md:22]: good shout! just added that `tags` are supported in 1.9+ under the `config` value:
Screenshot 2024-12-23 at 11 43 39

File: website/docs/reference/resource-configs/tags.md (hunk 4 above)
Comment (mirnawong1) [reply website/docs/reference/resource-configs/tags.md]: thanks @theyostalservice ! it does work for models (see screenshot). 
i tried seeing if it worked for saved queries, but i keep getting errors. can you tell me the specific command a user can use to run save queries with a specific tag? `dbt sl saved-queries --select tag:santi`, `dbt sl export --sq-name--select tag:santi`, `dbt sl export --select tag:santi`, `dbt sl saved-queries --select tag:santi` didn't work for me?


Screenshot 2024-12-23 at 12 03 26

File: website/docs/reference/resource-configs/tags.md [hunk 6]
```diff
@@ -164,10 +198,51 @@ seeds:
...
+saved_queries:
+  - name: test_saved_query
+    description: "{{ doc('saved_query_description') }}"
+    label: Test saved query
+    config:
+      tags: 
+        - order_metrics
+        - hourly
+```
+
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/tags.md]: add what command to use to list sq with tag name

File: website/docs/reference/resource-configs/tags.md [hunk 7]
```diff
@@ -164,10 +200,65 @@ seeds:
 
 </File>
 
+### Apply tags to saved queries
+
+<VersionBlock lastVersion="1.8">
```
Comment (nataliefiann) [inline website/docs/reference/resource-configs/tags.md:205]: Hiya @mirnawong1 

This is non-blocking suggestion. Would this benefit from an info callout? It may encourage the user to upgrade?

Kind Regards
Natalie

File: website/docs/reference/resource-configs/tags.md [hunk 8]
```diff
@@ -164,10 +200,65 @@ seeds:
...
+
+Applying tags to saved queries is only available in dbt Core versions 1.9 and later.
```
Comment (nataliefiann) [inline website/docs/reference/resource-configs/tags.md:209]: Hiya @mirnawong1

This is non-blocking suggestion. Would this benefit from an info callout? It may encourage the user to upgrade?

Kind Regards
Natalie

File: website/docs/reference/resource-configs/tags.md [hunk 9]
```diff
@@ -164,10 +200,65 @@ seeds:
...
+```yml
+[saved-queries](/docs/build/saved-queries):
+  jaffle_shop:
+    customer_order_metrics:
+      +tags: order_metrics
+```
+
+</File>
+
+Then, run part of your project like so:
```
Comment (nataliefiann) [inline website/docs/reference/resource-configs/tags.md]: Hiya @mirnawong1 

Apologies for the silly question - which part of the project would the user need to run? (ie the first part, last etc)? And does it matter which part of their project they need to run? Is this something dbt will automatically know?

Kind Regards
Natalie

**Review by nataliefiann — APPROVED** [review]
Hiya @mirnawong1 

Thanks for creating this PR. I've approved this for you. I've left a non-blocking suggestion and one question.

Kind Regards
Natalie

File: website/docs/reference/resource-configs/tags.md (hunk 8 above)
Comment (mirnawong1) [reply website/docs/reference/resource-configs/tags.md:209]: good shout, will add!

File: website/docs/reference/resource-configs/tags.md (hunk 9 above)
Comment (mirnawong1) [reply website/docs/reference/resource-configs/tags.md]: oo i'll clarify that they can run this command  to find the resources with a specific tag

Comment (mirnawong1) [reply website/docs/reference/resource-configs/tags.md]: changed it to this, good shout @nataliefiann !
Screenshot 2024-12-24 at 12 21 16
